app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala

```diff
@@ -72,49 +74,111 @@ class IncomeUpdateCalculatorController @Inject()(incomeService: IncomeService,
-            case (TaiSuccessResponseWithPayload(taxCodeIncomes: Seq[TaxCodeIncome]), Some(employment)) =>
-              val taxCodeIncomeSource = taxCodeIncomes.find(_.employmentId.contains(id)).
-                getOrElse(throw new RuntimeException(s"Income details not found for employment id $id"))
-              val isPension = taxCodeIncomeSource.componentType == PensionIncome
-              Ok(views.html.incomes.estimatedPayLandingPage(employment.name, id, isPension))
+          employmentDetails match {
+            case Some(employment) =>
+
+              val incomeType = incomeTypeIdentifier(employment.receivingOccupationalPension)
+
+              journeyCache(cacheMap = Map(
+                UpdateIncome_NameKey -> employment.name,
+                UpdateIncome_IdKey -> id.toString,
+                UpdateIncome_IncomeTypeKey -> incomeType))
+
+              if (estimatedPayCompletion) {
+                Redirect(routes.IncomeUpdateCalculatorController.duplicateSubmissionWarningPage())
+              } else {
+                Redirect(routes.IncomeUpdateCalculatorController.estimatedPayLandingPage())
+              }
             case _ => throw new RuntimeException("Not able to find employment")
           }
         }
   }
 
+  private def determineViewModel(incomeType: String, employmentName: String, newValue: Int): DuplicateSubmissionEstimatedPay = {
+    if (incomeType == TaiConstants.IncomeTypePension) {
+      DuplicateSubmissionPensionViewModel(employmentName, newValue)
+    } else {
+      DuplicateSubmissionEmploymentViewModel(employmentName, newValue)
+    }
+  }
+
+  def duplicateSubmissionWarningPage(): Action[AnyContent] = authorisedForTai(personService).async { implicit user =>
+    implicit person =>
+      implicit request =>
+
+      journeyCacheService.mandatoryValues(UpdateIncome_NameKey, UpdateIncome_IdKey, UpdateIncome_ConfirmedNewAmountKey, UpdateIncome_IncomeTypeKey) map { mandatoryValues =>
+        val incomeName :: incomeId :: newAmount :: incomeType :: Nil = mandatoryValues.toList
+
+        val vm = determineViewModel(incomeType, incomeName, newAmount.toInt)
+        Ok(views.html.incomes.duplicateSubmissionWarning(
+          DuplicateSubmissionWarningForm.createForm, vm, incomeId.toInt)
+          )
+        }
+  }
+
+  def submitDuplicateSubmissionWarning: Action[AnyContent] = authorisedForTai(personService).async { implicit user =>
+    implicit person =>
+      implicit request =>
+
+      journeyCacheService.mandatoryValues(UpdateIncome_NameKey, UpdateIncome_IdKey, UpdateIncome_ConfirmedNewAmountKey,  UpdateIncome_IncomeTypeKey) flatMap { mandatoryValues =>
+        val incomeName :: incomeId :: newAmount :: incomeType :: Nil = mandatoryValues.toList
+
+        DuplicateSubmissionWarningForm.createForm.bindFromRequest.fold(
+          formWithErrors => {
+            val vm = determineViewModel(incomeType, incomeName, newAmount.toInt)
+            Future.successful(BadRequest(views.html.incomes.
```

Comment:
I see what's newAmount here but its not clear from just reading the function parameters